nipype/utils/filemanip.py

```diff
@@ -425,6 +426,18 @@ def copyfiles(filelist, dest, copy=False, create_new=False):
             newfiles.insert(i, destfile)
     return newfiles
 
+def mkdirp(folder):
+    """
+    Equivalent to bash's mkdir -p
+    """
+    try:
+        os.makedirs(folder)
```

Comment:
on py3 we should just use `os.makedirs(exist_ok=True)`, (also see bug that was fixed here:  https://bugs.python.org/issue21082)

for py2, i'm still worried about race conditions, given that this is something people can start using. how about, just make this a hidden function for the moment and use it in the extension?